packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote_data_sources/interceptors/auth_interceptor.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'dart:convert';
+
+import 'package:dio/dio.dart';
+
+/// Interceptors are a simple way to intercept and modify http requests globally
+/// before they are sent to the server. That allows us to configure
+/// authentication tokens, add logs of the requests,
+/// add custom headers and much more.
+/// Interceptors can perform a variety of implicit tasks, from authentication
+/// to logging, for every HTTP request/response. Without interception, we will
+/// have to implement these tasks explicitly for each HttpClient method call.
+/// The AuthInterceptor will inject a Token in headers.
+/// This will allow us to remain logged in on the server side or check
+/// if the token exists and allow further REST api calls.
+/// Here is an example of AuthInterceptor. You have to implement
+/// your own logic, regarding needs of your application.
+class AuthInterceptor extends Interceptor {
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    // TODO: implement onResponse
+    super.onResponse(response, handler);
+  }
+
+  @override
+  Future<void> onRequest(
+      RequestOptions options, RequestInterceptorHandler handler) async {
+    final accessToken = await getToken();
+    if (accessToken != null) {
+      options.headers['Authorization'] = 'Bearer $accessToken';
+    }
+  //  Add your logic here
+    super.onRequest(options, handler);
+  }
+
+  @override
+  Future<void> onError(DioError err, ErrorInterceptorHandler handler)async {
+    if(err.response?.statusCode == 401){
+      final Map<String, dynamic> errData = jsonDecode(err.response.toString());
+      if(errData['details'] != 'Bad credentials'){
```

Comment:
@RomanovaPrime, @DDavidPrime do we really need this check for the string `Bad credentials` ?

Reply:
Deleted. I had took it from an example I used, but now I realize it depends on our API.